packages/babel-parser/src/parser/lval.js

```diff
@@ -19,6 +19,12 @@ import { isStrictBindReservedWord } from "../util/identifier";
 import { NodeUtils } from "./node";
 import { type BindingTypes, BIND_NONE } from "../util/scopeflags";
 
+const unwrapParenthesizedExpression = (node: Node) => {
```

Comment:
this is needed to handle nested parenthesis when `createParenthesizedExpressions` is enabled. i.e. `[(((x)))]`

Reply:
For nested parenthesis, `toAssignable` will be called with _parenthesizee_ recursively. Therefore we may check if `node` is one of `Identifier`, `MemberExpression` or `ParenthesizedExpression`.